src/Compilers/Core/Portable/Operations/ControlFlowGraphBuilder.cs

```diff
@@ -1006,12 +1006,55 @@ public override IOperation DefaultVisit(IOperation operation, int? captureIdForR
             throw ExceptionUtilities.Unreachable;
         }
 
-        #region PROTOTYPE(dataflow): Naive implementation that simply clones nodes and erases SemanticModel, likely to change
         internal override IOperation VisitNoneOperation(IOperation operation, int? captureIdForResult)
         {
-            return Operation.CreateOperationNone(semanticModel: null, operation.Syntax, operation.ConstantValue, () => VisitArray(operation.Children.ToImmutableArray()), operation.IsImplicit);
+            if (_currentStatement == operation)
+            {
+                VisitNoneOperationStatement(operation);
+                return null;
+            }
+            else
+            {
+                return VisitNoneOperationExpression(operation, captureIdForResult);
+            }
+        }
+
+        private void VisitNoneOperationStatement(IOperation operation)
+        {
+            Debug.Assert(_currentStatement == operation);
+            foreach (var child in operation.Children)
+            {
+                VisitStatement(child);
+            }
+        }
+
+        private IOperation VisitNoneOperationExpression(IOperation operation, int? captureIdForResult)
```

Comment:
>int? captureIdForResult [](start = 78, length = 23)

I think this parameter should be removed. Only operations that themselves represent a control flow are allowed to use this value and only for the final result.